fe/fe-core/src/main/java/com/starrocks/sql/analyzer/AnalyzeStmtAnalyzer.java

```diff
@@ -187,8 +186,16 @@ private void analyzeAnalyzeTypeDesc(ConnectContext session, AnalyzeStmt statemen
                         Double.parseDouble(properties.get(StatsConstants.HISTOGRAM_SAMPLE_RATIO)));
 
                 if (sampleRows < Config.statistic_sample_collect_rows && totalRows != 0) {
-                    properties.put(StatsConstants.HISTOGRAM_SAMPLE_RATIO,
-                            String.valueOf(BigDecimal.valueOf((double) sampleRows / (double) totalRows)
+                    if (Config.statistic_sample_collect_rows > totalRows) {
+                        properties.put(StatsConstants.HISTOGRAM_SAMPLE_RATIO, "1");
+                    } else {
+                        properties.put(StatsConstants.HISTOGRAM_SAMPLE_RATIO, String.valueOf(
+                                BigDecimal.valueOf((double) Config.statistic_sample_collect_rows / (double) totalRows)
+                                        .setScale(8, RoundingMode.HALF_UP).doubleValue()));
+                    }
```

Comment:
Why use `Config.statistic_sample_collect_rows` rather than `Config.histogram_max_sample_row_count` to calculate histogram sample ratio?

Reply:
Config.statistic_sample_collect_rows is a lower bound on the number of sampled rows, which applies to all sampling statistics. Config.histogram_max_sample_row_count is the upper bound of histogram sampling, and the maximum number of rows of the configuration is extracted